x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/profile/SuggestProfilesRequest.java

```diff
@@ -66,6 +83,96 @@ public ActionRequestValidationException validate() {
         if (size < 0) {
             validationException = addValidationError("[size] parameter cannot be negative but was [" + size + "]", validationException);
         }
+        if (hint != null) {
+            validationException = hint.validate(validationException);
+        }
         return validationException;
     }
+
+    public static class Hint implements Writeable {
+        @Nullable
+        private final List<String> uids;
+        @Nullable
+        private final Map<String, List<String>> labels;
+
+        public Hint(List<String> uids, Map<String, Object> labelsInput) {
+            this.uids = uids == null ? null : List.copyOf(uids);
+            // Convert the data type. Business logic, e.g. single key, is enforced in request validation
+            if (labelsInput != null) {
+                final HashMap<String, List<String>> labels = new HashMap<>();
+                for (Map.Entry<String, Object> entry : labelsInput.entrySet()) {
+                    final Object value = entry.getValue();
+                    if (value instanceof String) {
+                        labels.put(entry.getKey(), List.of((String) value));
+                    } else if (value instanceof final List<?> listValue) {
+                        final ArrayList<String> values = new ArrayList<>();
+                        for (Object v : listValue) {
+                            if (v instanceof final String stringValue) {
+                                values.add(stringValue);
+                            } else {
+                                throw new IllegalArgumentException("[labels] hint supports either string value or list of strings");
+                            }
+                        }
+                        labels.put(entry.getKey(), List.copyOf(values));
+                    } else {
+                        throw new IllegalArgumentException("[labels] hint supports either string or list of strings as its value");
```

Comment:
I think throwing `ElasticsearchParseException` is used in general for these, but it's mostly equivalent with `IllegalArgumentException`.

Reply:
Updated